source/AStar.cpp

```diff
@@ -72,22 +72,28 @@ AStar::CoordinateList AStar::Generator::findPath(Vec2i source_, Vec2i target_)
 {
     Node *current = nullptr;
     NodeSet openSet, closedSet;
-    openSet.insert(new Node(source_));
+    openSet.reserve(100);
+    closedSet.reserve(100);
+    openSet.push_back(new Node(source_));
 
     while (!openSet.empty()) {
-        current = *openSet.begin();
-        for (auto node : openSet) {
+        auto current_it = openSet.begin();
+        current = *current_it;
+
+        for (auto it = openSet.begin(); it != openSet.end(); it++) {
+            auto node = *it;
             if (node->getScore() <= current->getScore()) {
                 current = node;
+                current_it = it;
             }
         }
 
         if (current->coordinates == target_) {
             break;
         }
 
-        closedSet.insert(current);
-        openSet.erase(std::find(openSet.begin(), openSet.end(), current));
+        closedSet.push_back(current);
+        openSet.erase(current_it);
```

Comment:
We removed a search that was O(logN).
Since we saved the iterator earlier, basically this becomes O(1).